### DISCUSSION TITLE
Trying to connect to use a (self cert) to an internal https url but get 'failed to verify certificate: x509'

### DISCUSSION THREAD

**Discussion by G0nz0uk** [grafana/grafana-infinity-datasource#1082] [General]
Hello,

I have an internal URL on either of these that I need to show in Grafana via the Infinity json plugin:

URL
https://site.mydomain.com/nodes?format=json
or
https://192.168.200.243/nodes?format=json

The site has basic auth so just a username and password (at some point it will use LDAP)

I create an self certificate with the fqdn site.mydomain.com and added to our DNS servers and I can ping site.mydomain.com and get 192.168.200.243.

Now if I try in Grafana using the Infinity plugin I get:

`error while performing the infinity query. error getting response from url https://site.mydomain.com/nodes?format=json. no response received. Error: Get "https://192.168.200.243/nodes?format=jsonhttps%3A%2F%2Fsite.mydomain.com%2Fnodes%3Fformat%3Djson": tls: failed to verify certificate: x509: cannot validate certificate for 192.168.200.243 because it doesn't contain any IP SANs`

Please help, I'd love to get this working?

I was thing do I need to create a public fqdns and cert, but this is in dev and I just want to test this json works before I buy a proper name.

Using another JSON plugin on the Grafana site it works with that one:
\
<img width="707" alt="image" src="https://github.com/user-attachments/assets/ca615e0b-f409-41c8-8655-ddb1c4a9332d">

![image](https://github.com/user-attachments/assets/e43a732d-f1ff-4eb1-822b-b94d5df06bd2)

Thanks

**Comment by G0nz0uk** [comment]
Hi, I was just wondering if there is any update on this or can I do any more testing to help?

**Comment by Houlistonm** [comment]
@G0nz0uk Our team had this issue running on kubernetes.
Our fix was to load the CA for the selfsigned certs into the system CA truststore that golang uses.
Search the internet for GOLANG SSL truststore.
